gunicorn/arbiter.py

```diff
@@ -495,6 +495,17 @@ def manage_workers(self):
             (pid, _) = workers.pop(0)
             self.kill_worker(pid, signal.SIGTERM)
 
+        if self.pidfile is None and self.cfg.pidfile is not None and self.cfg.delay_pidfile:
+            worker_values = list(self.WORKERS.values())
+            booted = True
+            for worker in worker_values:
+                if not worker.booted:
```

Comment:
Also the fact that we're doing this in the manage_workers method which will get called on a regular basis by the arbiter. It's surprising to not have a loop. I wondered what would happen the first time this ran and none of the workers had successfully booted.